pallets/pallet-bank/src/tests.rs

```diff
@@ -601,3 +603,77 @@ fn can_force_transfer() {
 			assert!(Bank::check_total_issuance());
 		});
 }
+
+#[test]
+fn can_check_fund_unlock_at() {
+	MockGenesisConfig::with_balances(vec![(ALICE, 1_000)]).build().execute_with(|| {
+		// Check if it works with stake function.
+		let unlock_block = System::block_number() + StakePeriod::get();
+		assert_ok!(Bank::stake_funds(RuntimeOrigin::signed(ALICE), 100));
+
+		// Verify
+		assert_eq!(unlock_block, Bank::fund_unlock_at(ALICE, 1));
+
+		// Check it works with `AccountWithUnlockedFund` stores with multiple lock id.
+		AccountWithUnlockedFund::<Runtime>::insert(10, vec![(ALICE, 2), (ALICE, 5), (ALICE, 6)]);
```

Comment:
you need to add more test data - more users and a mixture of users and lock ID s to fully test the search function.

1 block with fully mixed locks from 3 users
blocks with locks from another user with the same lock ID

Reply:
Also test the case the lock cannot be found.